src/effects/backends/audiounit/dlgaudiounit.mm

```diff
@@ -0,0 +1,159 @@
+            (__bridge NSURL*)cocoaViewInfo.get()->mCocoaAUViewBundleLocation;
+    if (viewBundleLocation == nil) {
+        return "Cannot create UI of Audio Unit " + name +
+                " without view bundle path";
+    }
+
+    // We only use the first view as in the Cocoa AU host example linked earlier
+    NSString* factoryClassName =
+            (__bridge NSString*)cocoaViewInfo.get()->mCocoaAUViewClass[0];
+    ;
+    if (factoryClassName == nil) {
+        return "Cannot create UI of Audio Unit " + name +
+                " without factory class name";
+    }
+
+    NSBundle* viewBundle = [NSBundle bundleWithURL:viewBundleLocation];
+    if (viewBundle == nil) {
+        return "Could not load view bundle of Audio Unit " + name;
+    }
+
+    Class factoryClass = [viewBundle classNamed:factoryClassName];
+    if (factoryClass == nil) {
+        return "Could not load view factory class from bundle of Audio Unit " +
+                name;
+    }
+
+    id<AUCocoaUIBase> factoryInstance = [[factoryClass alloc] init];
+    if (factoryInstance == nil) {
+        return "Could not instantiate factory for view of Audio Unit " + name;
+    }
+
+    NSView* view = [factoryInstance uiViewForAudioUnit:audioUnit withSize:size];
+    if (view == nil) {
+        return "Could not instantiate view of Audio Unit " + name;
+    }
+
+    qDebug() << "Successfully loaded UI of Audio Unit" << name;
+    return view;
+}
+
+}; // anonymous namespace
+
+DlgAudioUnit::DlgAudioUnit(const AudioUnitManager& manager) {
+    QString name = manager.getName();
+
+    setWindowTitle(name);
+
+    // See
+    // https://lists.qt-project.org/pipermail/interest/2014-January/010655.html
+    // for why we need this slightly convoluted cast
+    NSView* dialogView = (__bridge NSView*)reinterpret_cast<void*>(winId());
+
+    // Style effect UI as a floating, but non-modal, HUD window
+    NSWindow* dialogWindow = [dialogView window];
+    [dialogWindow setStyleMask:NSWindowStyleMaskTitled |
+            NSWindowStyleMaskClosable | NSWindowStyleMaskUtilityWindow |
+            NSWindowStyleMaskHUDWindow];
+    [dialogWindow setLevel:NSFloatingWindowLevel];
+
+    auto result = createNativeUI(manager, size().toCGSize());
+
+    if (std::holds_alternative<NSView* _Nonnull>(result)) {
+        NSView* _Nonnull audioUnitView = std::get<NSView* _Nonnull>(result);
+        [dialogView addSubview:audioUnitView];
+
+        // Automatically resize the dialog to fit the view after layout
+        [audioUnitView setPostsFrameChangedNotifications:YES];
+        [[NSNotificationCenter defaultCenter]
+                addObserverForName:NSViewFrameDidChangeNotification
+                            object:audioUnitView
+                             queue:[NSOperationQueue mainQueue]
+                        usingBlock:^(NSNotification* notification) {
+                            NSView* audioUnitView =
+                                    (NSView*)notification.object;
```

Comment:
Small implementation note: We deliberately extract the view from the notification again to avoid creating a potential retain cycle by capturing the view again